### DISCUSSION TITLE
How to scale anycable on cloud?

### DISCUSSION THREAD

**Discussion by wrzepecki** [anycable/anycable#145] [Q&A] [answered]
How should it looks like when I want to scale anycable for more than one instance? Should I scale websocket server with gRPC in this same container or only websocket servers that will be pointing to one gRPC? How anycable will be know to which websocket server use to send stream? or maybe it will send to stream to all ws servers?

**Comment by Envek** [comment, marked as answer]
See the docs about scaling and load balancing: https://docs.anycable.io/deployment/load_balancing

 1. You can and should scale anycable-go and grpc instances independently. Depending on your connect-disconnect rate and number of websocket messages from clients to server their proportion will vary (server to clients messages doesn't use grpc instances)
 2. Your application will broadcast messages to all WS servers and these will re-transmit them to interested clients only.
